lms/views/helpers/frontend_app.py

```diff
@@ -12,9 +12,12 @@ def configure_grading(request, js_config):
     Note that this is entirely distinct from Canvas Speedgrader, which provides
     its own UI.
     """
+
+    lis_result_sourcedid_svc = request.find_service(name="lis_result_sourcedid")
+
     if (
         request.lti_user.is_instructor
-        and _is_assignment_gradable(request)
+        and lis_result_sourcedid_svc.is_assignment_gradable(request)
```

Comment:
Again there's a functional change here -- previously only `lis_outcome_service_url` is required but now an assignment will be considered ungradeable if any of `lis_outcome_service_url`, `lis_result_sourcedid`, `context_id` or `resource_link_id` are missing. Again, AFAIK it doesn't make a difference -- at least in the cases of Blackboard, Moodle and Brightspace they _do_ send all these params.